### PR TITLE
feat: empty cart view

### PR DESCRIPTION
**Pull request by kibo-susantasamanta**

### CODE REVIEW

File: components/page-templates/CartTemplate/CartTemplate.tsx [hunk 1]
```diff
@@ -224,37 +226,60 @@ const CartTemplate = (props: CartTemplateProps) => {
...
+                  onClick={handleGotoCheckout}
+                >
+                  {t('go-to-checkout')}
+                </Button>
+              </Stack>
+            </OrderSummary>
+          </Grid>
+        </>
+      ) : (
+        <>
```
Comment (TriptiVermaXP) [inline components/page-templates/CartTemplate/CartTemplate.tsx:264]: Empty cart is missing in storybook.

**Review by TriptiVermaXP — APPROVED** [review]
Overall looks good. Please fix one minor comment.

File: components/page-templates/CartTemplate/CartTemplate.tsx [hunk 2]
```diff
@@ -224,37 +226,60 @@ const CartTemplate = (props: CartTemplateProps) => {
...
+          <Typography variant="h4" fontWeight={'bold'}>
+            {t('cart:empty-cart-message')}
+          </Typography>
+          <Box maxWidth="23.5rem">
+            <Link href="/" passHref>
+              <MuiLink href="/">
```
Comment (TriptiVermaXP) [inline components/page-templates/CartTemplate/CartTemplate.tsx:270]: Do we need to give href in MuiLink if its present in Link?

File: components/page-templates/CartTemplate/CartTemplate.tsx (hunk 1 above)
Comment (sss-bct) [reply components/page-templates/CartTemplate/CartTemplate.tsx]: Done

File: components/page-templates/CartTemplate/CartTemplate.tsx (hunk 2 above)
Comment (sss-bct) [reply components/page-templates/CartTemplate/CartTemplate.tsx:270]: Yes. to test href attribute.

**Review by TriptiVermaXP — APPROVED** [review]

File: components/page-templates/CartTemplate/CartTemplate.spec.tsx [hunk 3]
```diff
@@ -48,4 +50,21 @@ describe('[components] CartTemplate', () => {
...
+    const cartParams = { ...Common.args }
+    if (cartParams.cart) cartParams.cart.items = []
+
+    const { user, router } = setup(cartParams as CartTemplateProps)
+    await waitFor(async () => {
+      const emptyCartSubTitle = screen.getByText(/cart:empty-cart-message/i)
+
+      expect(emptyCartSubTitle).toBeVisible()
+    })
+    const shopNowButton = screen.getByText(/common:shop-now/i)
```
[the remark this answers is not in the document]
Comment (sss-bct) [reply components/page-templates/CartTemplate/CartTemplate.spec.tsx]: Used role.

**Review by sushantj2050dev — APPROVED** [review]

**Review by kibo-kevinwatts — APPROVED** [review]

File: components/page-templates/CartTemplate/CartTemplate.tsx [hunk 4]
```diff
@@ -224,37 +226,60 @@ const CartTemplate = (props: CartTemplateProps) => {
...
-              sx={{ ...styles.checkoutButtonStyle }}
-              fullWidth
-              onClick={handleGotoCheckout}
-            >
-              {t('go-to-checkout')}
-            </Button>
-          </Stack>
-        </OrderSummary>
-      </Grid>
+      {cart?.items?.length ? (
```
Comment (kibo-chandradeeptalaha) [inline components/page-templates/CartTemplate/CartTemplate.tsx:229]: Can we use && instead of ternary ?

Comment (sss-bct) [reply components/page-templates/CartTemplate/CartTemplate.tsx]: Done.

**Review by kibo-chandradeeptalaha — APPROVED** [review]

**Pull request merged**